src/app/my-appliances/components/deployments/deployment/deployment.component.ts

```diff
@@ -73,16 +73,27 @@ export class DeploymentComponent implements OnInit, OnDestroy {
     // information was over 10 minutes ago, then automatically kick off a
     // HEALTH_CHECK task
     initializeAutomaticHealthCheck() {
+        const latest_task = this.deployment.latest_task;
+        if (latest_task.status !== 'PENDING' && latest_task.status !== 'PROGRESSING' && latest_task.action !== 'DELETE'
+            && latest_task.result.instance_status !== 'unknown' && latest_task.result.instance_status !== 'deleted'
+            && latest_task.result.instance_status !== 'not_found') {
+
+            const addedMoment = moment(latest_task.added);
+            if (addedMoment.isBefore(moment().subtract(AUTOMATIC_HEALTH_CHECK_MINUTES, 'minutes'))) {
+                this.runTask(this.deployment, 'HEALTH_CHECK');
+            }
+        }
+
         return this.defaultCreds.subscribe(credentials => {
-            if (credentials) {
+            /*if (credentials) {
```

Comment:
I think the original reason that this was done on credential changes is because ngOnInit is only called once when the component is first created: https://angular.io/guide/lifecycle-hooks
This means that the health check does not rerun even if the credentials change. Therefore, the more correct way would be to do this on credentials change. In addition though, we would need to combine this with a timer too, so that every X seconds, the health check is retriggered. This can be done by using an RxJs Timer (http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#static-method-timer)